### PR TITLE
docs: use __init__ method instead of constructor

### LINKED ISSUE #4086: constructor, initializer → __init__ method
There are several places in the documentation and comments where we use the words ‘constructor’ or ‘initializer’ to refer to the `__init__` method of a class. We should change the wording to explicitly say ‘`__init__` method’ instead.

Related to #2354.

### PR DESCRIPTION
**Pull request by ammarnajjar**
Resolves #4086

### CODE REVIEW

**Comment by Gallaecio** [comment]
You have successfully used double ` in many documentation changes, but some changes use a single ` instead, which is incorrect (it applies italics).

**Comment by ammarnajjar** [comment]
thanks for the review, I hope I got it right this time 😅

File: docs/topics/loaders.rst [hunk 1]
```diff
@@ -265,7 +265,7 @@ There are several ways to modify Item Loader context values:
       loader.context['unit'] = 'cm'
 
 2. On Item Loader instantiation (the keyword arguments of Item Loader
-   constructor are stored in the Item Loader context)::
+   ``__init__`` methodare stored in the Item Loader context)::
```
Comment (Gallaecio) [inline docs/topics/loaders.rst]:
```suggestion
   ``__init__`` method are stored in the Item Loader context)::
```

I’ve seen this issue in a few more replacements below, where the character right after ‘method’ (a space, a dot, a comma) was accidentally removed. Could you please take a look?

Comment (ammarnajjar) [reply docs/topics/loaders.rst]: 🤦‍♂ of course

Comment (ammarnajjar) [reply docs/topics/loaders.rst]: @Gallaecio thanks for your patience, change pushed.

File: tests/test_utils_misc/__init__.py [hunk 2]
```diff
@@ -109,11 +109,11 @@ def _test_with_crawler(mock, settings, crawler):
...
-        # Check usage of correct constructor using four mocks:
-        #   1. with no alternative constructors
-        #   2. with from_settings() constructor
-        #   3. with from_crawler() constructor
-        #   4. with from_settings() and from_crawler() constructor
+        # Check usage of correct __init__ method using four mocks:
+        #   1. with no alternative __init__ methods
+        #   2. with from_settings() __init__ method
+        #   3. with from_crawler() __init__ method
+        #   4. with from_settings() and from_crawler() __init__ method
```
Comment (kmike) [inline tests/test_utils_misc/__init__.py]: I think here "constructor" is replaced with `__init__` incorrectly; "constructor" here is a single word for from_crawler/from_settings/etc., not as a synonym to `__init__`.

Comment (ammarnajjar) [reply tests/test_utils_misc/__init__.py]: so should one leave it as is "constructor" or use “factory method” instead as @Gallaecio  mentioned in https://github.com/scrapy/scrapy/pull/4088#pullrequestreview-305189732?

Comment (kmike) [reply tests/test_utils_misc/__init__.py]: Yeah, I think so; no strong opinion about constructor/factory methods, I was always thinking about them as of "alternative constructors", so comment before the change as-is looks fine to me.

Comment (ammarnajjar) [reply tests/test_utils_misc/__init__.py]: @kmike thanks for the review, 7a84a4b is pushed.

File: scrapy/extensions/feedexport.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def __init__(self, uri, access_key=None, secret_key=None, acl=None):
                 warnings.warn(
                     "Initialising `scrapy.extensions.feedexport.S3FeedStorage` "
                     "without AWS keys is deprecated. Please supply credentials or "
-                    "use the `from_crawler()` constructor.",
+                    "use the `from_crawler()` ``__init__`` method.",
```
Comment (kmike) [inline scrapy/extensions/feedexport.py]: it should be kept as "constructor" here

File: sep/sep-009.rst [hunk 4]
```diff
@@ -55,7 +55,7 @@ singletons members of that object, as explained below:
      ``STATS_CLASS`` setting)
    - **crawler.log**: Logger class with methods replacing the current
      ``scrapy.log`` functions. Logging would be started (if enabled) on
-     ``Crawler`` constructor, so no log starting functions are required.
+     ``Crawler`` ``__init__`` method, so no log starting functions are required.
```
Comment (kmike) [inline sep/sep-009.rst]:
```suggestion
     ``Crawler`` instantiation, so no log starting functions are required.
```

**Pull request merged**